### ISSUE TITLE
Incorrect SQL query generated from linq expression

### ISSUE THREAD

**Issue by Unrefraction** [Azure/azure-documentdb-net#38]
Hi,

I've been studying console app example, and I tried to select all families which have children that are in fifth grade.

So here is my linq query:
```
var list = client.CreateDocumentQuery<Family>(documentCollection.DocumentsLink)
             .SelectMany(family => family.Children
             .Where(x => x.Grade == 5)
             .Select(child => new
             {
                 family = family.Id,
                 child = child.FirstName
             })).ToList();
```

But resultset contains only 'family' values whereas 'child' is null.
I'd checked generated SQL query and got this:

```
SELECT VALUE {family: root.id, child: root.FirstName} FROM root JOIN x IN root.Children WHERE (x.Grade = 5) 
```

so FirstName value takes from 'root' (Family), not from 'x'

Is it an issue in query builder or i'm just building expression in a wrong way?

**Comment by ryancrawcour** [comment]
Thank you for raising this. This has been confirmed as a bug with the LINQ provider and we will fix it in a future release. 

As a workaround, please use SQL instead of the LINQ provider when selecting anonymous types like this. 

Thanks.

**Comment by ryancrawcour** [comment]
This has been fixed with the release of SDK 1.3.0
